### ISSUE TITLE
Corrupted matrix when using na.exclude

### ISSUE THREAD

**Issue by jwdink** [decisionpatterns/sticky#2]
E.g., 

```
my_mat <- matrix(data = c(1:5,NA), nrow = 3)

my_sticky_mat <- sticky::sticky(my_mat)

na.exclude(my_mat)

na.exclude(my_sticky_mat)
```

**Comment by ctbrown** [comment]
The code from @jwdink produces the following error:

> Error in class(x) <- setdiff(class(x), "sticky") : 
>   invalid to set the class to matrix unless the dimension attribute is of length 2 (was 0)
> 
> ```
> traceback()
> 5: .deinit(x) at sticky.R#52
> 4: unstick(x)
> 3: print(unstick(x), ...) at print.R#26
> 2: print.sticky(x)
> 1: print(x)
> ```

This:
- occurs in `.deinit(x)` 
- appears to be more a problem with `stats::na.*` not respecting the class attribute vector than with the sticky package.  As `stats::na.*` functions are generic, this seems like it would be an easy, straight-forward fix, but it might be better to file a bug with `stats` instead.  See how `na.exclude` is not keeping the `dim` attribute though it is class matrix.

```
my_mat <- matrix(data = c(1:5,NA), nrow = 3)

my_sticky_mat <- sticky(my_mat)

x1 <- na.exclude(my_mat)
dim(x1)  # 2 2

x2 <- na.exclude(my_sticky_mat)
dim(x2)  # NULL

> class(my_sticky_mat)
[1] "sticky" "matrix"
> class(x2)
[1] "sticky" "matrix"
```

It would seem wrong to change the inheritance order here.  It seems the short-term solution is to define `na.exclude.sticky` as a work-around.

**Comment by ctbrown** [comment]
Looking a little more into it, this is not a problem with `na.exclude`. The following does not work.

```
m <- matrix( 1:6, nrow=3 )
m[1:2,1:2]

sm <- sticky(m)
sm[1:2,]
```

The package seems to fail for subsets of matrices.

**Comment by ctbrown** [comment]
Fixed #2.. 
It is available on github now and will be available as sticky-0.4.0 on CRAN once all the checks are complete.
